src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java

```diff
@@ -383,4 +387,48 @@ public void testFindMethodWithManyToOneRelationRelatedObjectIsNull() {
         entityManager.getTransaction().commit();
     }
 
+    @Test
+    @DisplayName("Test lazy loading. Get list of related objects out of transaction")
+    public void testLazyLoading() {
+
+        PersonWithoutGeneratedValue person = new PersonWithoutGeneratedValue();
+        person.setFirstName("Quentin");
+        person.setLastName("Tarantino");
+        person.setId(40L);
+        person.setBirthday(LocalDateTime.of(1963, Month.MARCH, 27, 10, 0, 0).toLocalDate());
+
+        NoteWithoutGeneratedValue firstNote = new NoteWithoutGeneratedValue();
+        firstNote.setId(51L);
+        firstNote.setBody("Test note #1");
+        firstNote.setPerson(person);
+
+        NoteWithoutGeneratedValue secondNote = new NoteWithoutGeneratedValue();
+        secondNote.setId(52L);
+        secondNote.setBody("Test note #2");
+        secondNote.setPerson(person);
+
+        person.addNote(firstNote);
+        person.addNote(secondNote);
+
+        entityManager.getTransaction().begin();
+        entityManager.persist(person);
+        entityManager.getTransaction().commit();
+
+        EntityManagerImpl otherEntityManager = new EntityManagerImpl(dataSource);
+        otherEntityManager.getTransaction().begin();
+        var selectedPerson = otherEntityManager.find(PersonWithoutGeneratedValue.class, person.getId());
+        otherEntityManager.getTransaction().commit();
+        otherEntityManager.close();
+
+        assertEquals(person.getFirstName(), selectedPerson.getFirstName());
+        assertEquals(person.getLastName(), selectedPerson.getLastName());
+        assertEquals(person.getBirthday(), selectedPerson.getBirthday());
+        assertThrows(LazyInitializationException.class, () -> selectedPerson.getNotes().size());
```

Comment:
I am not sure why we expect here LazyInitializationException.class. In theory it should not throw exception at all as selectPerson.getNotes() has internal lazy list. When we call getNotes() it will produce sql queries and will fetch notes. Hence, our notes list will not be empty and should not throw any exception